### PR TITLE
Update lease should update LeaseStatusModifiedOn

### PR DESCRIPTION
**Pull request by AmudaPalani**
## Proposed changes

Update lease should update LeaseStatusModifiedOn

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: tests/acceptance/api_test.go [hunk 1]
```diff
@@ -1330,6 +1330,19 @@ func TestApi(t *testing.T) {
...
+				method: "GET",
+				url:    apiURL + fmt.Sprintf("/leases?principalId=test-user&accountId=%s", accountID),
+				json:   nil,
+			})
+
+			results := parseResponseArrayJSON(t, resp)
+			assert.Equal(t, 200, resp.StatusCode)
+			assert.Equal(t, 1, len(results), "one lease should be returned")
+			assert.Equal(t, "Active", results[0]["leaseStatus"])
+			assert.NotNil(t, results[0]["leaseStatusModifiedOn"])
```
Comment (nathanagood) [inline tests/acceptance/api_test.go:1344]: We're probably not doing this anywhere else, but do we want to add any different checks other than NotNil? Should it be formatted a certain way? Greater than today, etc.?

Comment (AmudaPalani) [reply tests/acceptance/api_test.go:1344]: @nathanagood I will add a check to verify leaseStatusModifiedOn date is equal to CreatedOn date.

File: pkg/lease/service.go [hunk 2]
```diff
@@ -91,6 +91,7 @@ func (a *Service) Save(data *Lease) error {
 	} else {
 		lastModifiedOn = data.LastModifiedOn
 		data.LastModifiedOn = &now
+		data.StatusModifiedOn = &now
```
Comment (eschwartz) [inline pkg/lease/service.go:94]: @AmudaPalani I don't think this is quite what we want.

The `LeaseStatusModifiedOn` field should only be updated if the `LeaseStatus` field changes.
Unfortunately, we have no way within the `Save()` method to know if the `LeaseStatus` field has been updated (we only have the new lease object, not the old one).

I'm not sure how best to approach this --we may need to set this timestamp  in some of the higher-level service methods (eg. `Create()` / `Delete()`). What do you think?

Comment (AmudaPalani) [reply pkg/lease/service.go:94]: This line 92-94 will be executed in two scenarios only:
a) Update a lease status from active to inactive (Decommission).
b) Update a lease status from inactive to active (Same account getting leased).
In both these scenarios, we want to update LeaseStatusModifiedOn.

**Pull request merged**